### PR TITLE
Fix OCR reasoning cleanup for unclosed/dangling <think> tags (follow-up to #708)

### LINKED ISSUE #914: OCR reasoning cleanup misses unclosed and dangling think tags in model output (follow-up to #708)
### Summary
Issue #708 improved OCR output by stripping think-tag reasoning blocks, but there is still an edge-case gap: unclosed opening think tags and dangling closing think tags can leak reasoning artifacts or malformed text into final OCR content.

This is a follow-up bug to #708 (closed), not a duplicate.

### Steps to reproduce
1. Configure OCR with a reasoning-capable vision model that may emit think tags.
2. Process content that includes one of these patterns:
   - Content <think>Unclosed reasoning
   - thinkingstuff </think> content <-- qwen3-vl
3. Inspect the final extracted OCR text returned by the OCR pipeline.

### Actual behavior
OCR output may include malformed reasoning remnants or fail to fully sanitize dangling/unbalanced think tags.

### Expected behavior
OCR sanitization should be robust to malformed tag pairs:
- If an opening think tag appears without a closing tag, remove the opening tag and everything after it.
- If a closing think tag appears without a valid preceding opening tag, treat preceding malformed reasoning text as discardable and keep only valid trailing content.
- Final output should be trimmed and contain only user-visible OCR text.

### Why this matters
Some reasoning models do not always emit perfectly balanced tags. Without robust cleanup, extracted OCR content can include non-document reasoning text, reducing quality and downstream metadata accuracy.

### Environment
OS: Linux
App: paperless-gpt
LLM provider/model: affected with reasoning-style vision models (specifically `qwen3-vl`)

### PR DESCRIPTION
**Pull request by MilanSchilling**
### Problem
Issue #708 added reasoning-tag stripping for OCR output, but malformed tag patterns were still not fully handled in all flows:
- Unclosed opening tag: `<think>...` with no `</think>`
- Dangling closing tag: `</think>` appearing without a valid opening tag

In addition, reasoning stripping logic existed in multiple places (OCR and app LLM paths), which risked behavior drift and inconsistent sanitization (kudos to @wickode).

This could leak reasoning artifacts into final text and produce different results depending on code path.

### What changed
- Centralized reasoning sanitization into a shared internal helper (`internal/textsanitize/StripReasoning`) as the single source of truth.
- Applied the shared sanitizer across both OCR and app LLM suggestion paths.
- Removed duplicated local OCR utility implementation and its duplicate test file after centralization.
- Removed duplicated app-local implementation and switched app tests to use the shared helper.
- Added/kept robust unit coverage for balanced and malformed <think>/</think> patterns in the shared sanitizer tests, including:
  - unclosed opening tag strips trailing reasoning content
  - dangling closing tag with trailing valid content preserves only valid content
  - balanced-tag behavior remains covered

### Before / After
**Before**
- `Content <think>Unclosed reasoning` → reasoning remnants could remain
- `thinkingstuff </think> content` → malformed output could persist

**After**
- `Content <think>Unclosed reasoning` → `Content`
- `thinkingstuff </think> content` → `content`
- OCR and app LLM flows now use identical sanitization behavior

### Testing
- Added shared sanitizer unit tests covering malformed-tag scenarios and existing balanced-tag behavior.
- Updated app test usage to validate through the shared sanitizer.
- Full test suite passes on this branch.

### Why this approach
Centralizing sanitization removes duplication and drift, ensures consistent behavior across OCR and non-OCR LLM outputs, and makes future fixes/test updates happen in one place. This gives more defensive handling for real-world model responses where reasoning tags may be malformed.

## Related issue
Fixes #914
(also follow-up context: #708)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved removal of embedded reasoning markers so multiple, unclosed, or dangling markers no longer leave artifacts; trailing whitespace is trimmed and outputs are cleaner across OCR and LLM result flows.

* **Tests**
  * Updated test coverage to validate the revised stripping behavior and edge cases, ensuring more reliable outputs.

### CODE REVIEW

File: internal/textsanitize/reasoning.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		openIdx := strings.Index(remaining, openThinkTag)
+		closeIdx := strings.Index(remaining, closeThinkTag)
+
+		switch {
+		case openIdx == -1 && closeIdx == -1:
+			builder.WriteString(remaining)
+			remaining = ""
+		case closeIdx != -1 && (openIdx == -1 || closeIdx < openIdx):
+			builder.Reset()
+			remaining = remaining[closeIdx+len(closeThinkTag):]
```
[the remark this answers is not in the document]
Comment (MilanSchilling) [reply internal/textsanitize/reasoning.go:31]: I disagree that this is a major potential issue: Some models might omit the initial opening `<think>` tag, but then print some reasoning  followed by a dangling closing `</think>` tag and the rest of the output. In my opinion it is very unlikely that a model would print a complete `<think> ... </think>` sequence but then omit a following opening `<think>` tag again.

**Comment by Itstommy10** [comment]
Hi @MilanSchilling , sorry to bother you, I'm having this problem. I'm using ollama
qwen3-vl:latest
but in the logs I see that the prompt ends in thinking.

{"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.530537676Z","message":{"role":"assistant","content":"","thinking":"_expiry"},"done":false}

17:28:50.530657 eth0 Out IP paperless-gpt.Home.48998 > ollama.Home.11434: Flags [.], ack 148372, win 1007, options [nop,nop,TS val 1619128064 ecr 905045559], length 0
E..4.j@.@.....
j..
..f,...J..4.............
`...5..7
17:28:50.557674 eth0 In IP ollama.Home.11434 > paperless-gpt.Home.48998: Flags [P.], seq 148372:148528, ack 1, win 842, options [nop,nop,TS val 905045586 ecr 1619128064], length 156
AND.....@.@.....
...
j,..f.4....J....J.7.....
5..R`...96
{"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.557619993Z","message":{"role":"assistant","content":"","thinking":"_date"},"done":false}


while from what I understand qwen3 expects me to go to content? precisely when I start an OCR I get a generic response like:


It seems like you're asking about the "best way to handle a problem," but you haven't specified **what kind of problem** you're referring to. Problem-solving strategies vary widely depending on the context—whether it's a technical issue, a personal challenge, a work-related dilemma, or something else.

**Comment by MilanSchilling** [comment]
> Hi @MilanSchilling , sorry to bother you, I'm having this problem. I'm using ollama qwen3-vl:latest but in the logs I see that the prompt ends in thinking.
> 
> {"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.530537676Z","message":{"role":"assistant","content":"","thinking":"_expiry"},"done":false}
> 
> 17:28:50.530657 eth0 Out IP paperless-gpt.Home.48998 > ollama.Home.11434: Flags [.], ack 148372, win 1007, options [nop,nop,TS val 1619128064 ecr 905045559], length 0 E..4.j@.@..... j.. ..f,...J..4............. `...5..7 17:28:50.557674 eth0 In IP ollama.Home.11434 > paperless-gpt.Home.48998: Flags [P.], seq 148372:148528, ack 1, win 842, options [nop,nop,TS val 905045586 ecr 1619128064], length 156 AND.....@.@..... ... j,..f.4....J....J.7..... 5..R`...96 {"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.557619993Z","message":{"role":"assistant","content":"","thinking":"_date"},"done":false}
> 
> while from what I understand qwen3 expects me to go to content? precisely when I start an OCR I get a generic response like:
> 
> It seems like you're asking about the "best way to handle a problem," but you haven't specified **what kind of problem** you're referring to. Problem-solving strategies vary widely depending on the context—whether it's a technical issue, a personal challenge, a work-related dilemma, or something else.

Hi @Itstommy10

I'm not sure if the problem you describe is directly related to the changes I propose in this PR here. I'm also running `qwen3-vl:8b` in ollama and I don't see the same error.

Maybe it's worth to open a new issue for the specific problem you're seeing.

**Comment by Itstommy10** [comment]
> > Hi @MilanSchilling , sorry to bother you, I'm having this problem. I'm using ollama qwen3-vl:latest but in the logs I see that the prompt ends in thinking.
> > {"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.530537676Z","message":{"role":"assistant","content":"","thinking":"_expiry"},"done":false}
> > 17:28:50.530657 eth0 Out IP paperless-gpt.Home.48998 > ollama.Home.11434: Flags [.], ack 148372, win 1007, options [nop,nop,TS val [1619128064](tel:1619128064) ecr 905045559], length 0 E..4.j@.@..... j.. ..f,...J..4............. `...5..7 17:28:50.557674 eth0 In IP ollama.Home.11434 > paperless-gpt.Home.48998: Flags [P.], seq 148372:148528, ack 1, win 842, options [nop,nop,TS val 905045586 ecr 1619128064], length 156 AND.....@.@..... ... j,..f.4....J....J.7..... 5..R`...96 {"model":"qwen3-vl:latest","created_at":"2026-03-06T16:28:50.557619993Z","message":{"role":"assistant","content":"","thinking":"_date"},"done":false}
> > while from what I understand qwen3 expects me to go to content? precisely when I start an OCR I get a generic response like:
> > It seems like you're asking about the "best way to handle a problem," but you haven't specified **what kind of problem** you're referring to. Problem-solving strategies vary widely depending on the context—whether it's a technical issue, a personal challenge, a work-related dilemma, or something else.
> 
> Hi @Itstommy10
> 
> I'm not sure if the problem you describe is directly related to the changes I propose in this PR here. I'm also running `qwen3-vl:8b` in ollama and I don't see the same error.
> 
> Maybe it's worth to open a new issue for the specific problem you're seeing.

Hi, first of all thank you for your availability. I think you're right, I just wanted confirmation that mine was a fairly isolated case. Seeing that you also use the same model, can I ask you if you use an ad hoc personalized gpt paperless prompt to perform the ocr? I can think it's mine that's causing the problem

**Comment by icereed** [comment]
Thank you so much for this contribution! 🙏 This change landed on `main` as part of the community bundle #990 and is now released in [v0.26.0 – "The Resilience Expansion"](https://github.com/icereed/paperless-gpt/releases/tag/v0.26.0), where you're credited in the release notes. Closing this PR since the change is included there — thanks again for making paperless-gpt better!
